lib/features/server_settings/data/network/server_settings_api.dart

```diff
@@ -0,0 +1,81 @@
+import 'package:jmap_dart_client/jmap/account_id.dart';
+import 'package:jmap_dart_client/jmap/jmap_request.dart';
+import 'package:server_settings/server_settings/get/get_server_settings_method.dart';
+import 'package:server_settings/server_settings/get/get_server_settings_response.dart';
+import 'package:server_settings/server_settings/server_settings_id.dart';
+import 'package:server_settings/server_settings/set/set_server_settings_method.dart';
+import 'package:server_settings/server_settings/tmail_server_settings.dart';
+import 'package:tmail_ui_user/features/base/mixin/handle_error_mixin.dart';
+import 'package:tmail_ui_user/features/server_settings/domain/exceptions/server_settings_exception.dart';
+
+class ServerSettingsAPI with HandleSetErrorMixin {
+  final HttpClient httpClient;
+
+  ServerSettingsAPI(this.httpClient);
+
+  Future<TMailServerSettings> getServerSettings(AccountId accountId) async {
+    final processingInvocation = ProcessingInvocation();
+
+    final jmapRequestBuilder = JmapRequestBuilder(httpClient, processingInvocation);
+
+    final getServerSettingsMethod = GetServerSettingsMethod(accountId);
+
+    final queryInvocation = jmapRequestBuilder.invocation(getServerSettingsMethod);
+
+    final capabilities = getServerSettingsMethod.requiredCapabilities;
+
+    final result = await (jmapRequestBuilder..usings(capabilities))
+      .build()
+      .execute();
+
+    final serverSettings = result.parse<GetServerSettingsResponse>(
+      queryInvocation.methodCallId,
+      GetServerSettingsResponse.deserialize
+    )?.list.firstOrNull;
+
+    if (serverSettings == null) {
+      throw NotFoundServerSettingsException();
+    }
+
+    return serverSettings;
+  }
+
+  Future<TMailServerSettings> updateServerSettings(
+    AccountId accountId, 
+    TMailServerSettings serverSettings
+  ) async {
+    final processingInvocation = ProcessingInvocation();
+
+    final setServerSettingsMethod = SetServerSettingsMethod(accountId)
+      ..addUpdatesSingleton({
+        ServerSettingsIdSingleton.serverSettingsIdSingleton.id: serverSettings
+      });
+
+    final jmapRequestBuilder = JmapRequestBuilder(httpClient, processingInvocation)
+      ..invocation(setServerSettingsMethod);
+
+    final getServerSettingsMethod = GetServerSettingsMethod(accountId)
+      ..addIds({ServerSettingsIdSingleton.serverSettingsIdSingleton.id});
+
+    final getServerSettingsInvocation = jmapRequestBuilder.invocation(
+      getServerSettingsMethod);
+
+    final capabilities = setServerSettingsMethod.requiredCapabilities;
+
+    final result = await (jmapRequestBuilder..usings(capabilities))
+      .build()
+      .execute();
+
+    final updatedServerSettings = result.parse<GetServerSettingsResponse>(
+      getServerSettingsInvocation.methodCallId,
+      GetServerSettingsResponse.deserialize
+    )?.list.firstOrNull;
+
+    if (updatedServerSettings == null) {
+      throw NotFoundServerSettingsException();
```

Comment:
IMO, we should throw `CanNotUpdateServerSettingsException`.

Reply:
Done